source/patterns/@aws-solutions-constructs/core/lib/opensearch-helper.ts

```diff
@@ -0,0 +1,306 @@
+import { Construct } from 'constructs';
+
+const MAXIMUM_AZS_IN_OPENSEARCH_DOMAIN = 3;
+
+export interface BuildOpenSearchProps {
+  readonly identitypool: cognito.CfnIdentityPool;
+  readonly userpool: cognito.UserPool;
+  readonly cognitoAuthorizedRoleARN: string;
+  readonly serviceRoleARN?: string;
+  readonly vpc?: ec2.IVpc;
+  readonly openSearchDomainName: string;
+  readonly clientDomainProps?: opensearch.CfnDomainProps,
+  readonly securityGroupIds?: string[]
+}
+
+export function buildOpenSearch(scope: Construct, props: BuildOpenSearchProps): [opensearch.CfnDomain, iam.Role] {
+  let subnetIds: string[] = [];
+  const constructDrivenProps: any = {};
+
+  // Setup the IAM Role & policy for the OpenSearch Service to configure Cognito User pool and Identity pool
+  const cognitoKibanaConfigureRole = createKibanaCognitoRole(scope, props.userpool, props.identitypool, props.openSearchDomainName);
+
+  if (props.vpc) {
+    subnetIds = retrievePrivateSubnetIds(props.vpc);
+
+    if (subnetIds.length > MAXIMUM_AZS_IN_OPENSEARCH_DOMAIN) {
+      subnetIds = subnetIds.slice(0, MAXIMUM_AZS_IN_OPENSEARCH_DOMAIN);
+    }
+
+    constructDrivenProps.vpcOptions = {
+      subnetIds,
+      securityGroupIds: props.securityGroupIds
+    };
+
+    // If the client did not submit a ClusterConfig, then we will create one
+    if (!props.clientDomainProps?.clusterConfig) {
+      constructDrivenProps.clusterConfig = createClusterConfiguration(subnetIds.length);
+    }
+  } else { // No VPC
+    // If the client did not submit a ClusterConfig, then we will create one based on the Region
+    if (!props.clientDomainProps?.clusterConfig) {
+      constructDrivenProps.clusterConfig = createClusterConfiguration(cdk.Stack.of(scope).availabilityZones.length);
+    }
+  }
+
+  const defaultCfnDomainProps = DefaultOpenSearchCfnDomainProps(props.openSearchDomainName, cognitoKibanaConfigureRole, props);
+  const finalCfnDomainProps = consolidateProps(defaultCfnDomainProps, props.clientDomainProps, constructDrivenProps);
+
+  const opensearchDomain = new opensearch.CfnDomain(scope, `OpenSearchDomain`, finalCfnDomainProps);
+
+  addCfnSuppressRules(opensearchDomain, [
+    {
+      id: "W28",
+      reason: `The OpenSearch Service domain is passed dynamically as as parameter and explicitly specified to ensure that IAM policies are configured to lockdown access to this specific OpenSearch Service instance only`,
+    },
+    {
+      id: "W90",
+      reason: `This is not a rule for the general case, just for specific use cases/industries`,
+    },
+  ]);
+
+  return [opensearchDomain, cognitoKibanaConfigureRole];
+}
+
+export function buildOpenSearchCWAlarms(scope: Construct): cloudwatch.Alarm[] {
+  const alarms: cloudwatch.Alarm[] = new Array();
+
+  // ClusterStatus.red maximum is >= 1 for 1 minute, 1 consecutive time
+  alarms.push(new cloudwatch.Alarm(scope, 'StatusRedAlarm', {
+    metric: new cloudwatch.Metric({
+      namespace: 'AWS/ES',
```

Comment:
Why still ES instead of OS?

Reply:
OpenSearch kept ES prefix instead of creating new OS